metadata-ingestion/src/datahub/ingestion/masking/bootstrap.py

```diff
@@ -0,0 +1,461 @@
+"""
+Bootstrap module for secret masking initialization.
+"""
+
+import logging
+import os
+import sys
+import threading
+from enum import Enum
+from typing import List, Optional
+
+from datahub.ingestion.masking.logging_utils import get_masking_safe_logger
+from datahub.ingestion.masking.masking_filter import (
+    SecretMaskingFilter,
+    install_masking_filter,
+    uninstall_masking_filter,
+)
+from datahub.ingestion.masking.secret_registry import SecretRegistry
+
+logger = get_masking_safe_logger(__name__)
+
+# Bootstrap state tracking
+_bootstrap_completed = False
+_bootstrap_error: Optional[Exception] = None
+_original_excepthook = None  # Track original exception hook for restoration
+_bootstrap_lock = threading.Lock()  # Thread safety for concurrent initialization
+
+
+class ExecutionContext(Enum):
+    """Execution context for DataHub ingestion."""
+
+    CLI = "cli"
+    UI_BACKEND = "ui_backend"
+    REMOTE_EXECUTOR = "remote"
+    SCHEDULED = "scheduled"
+    UNKNOWN = "unknown"
```

Comment:
I understand different context will have different sources of secret. So far, cli masks env vars. Remote executor will mask secrets from secret store, and so on.

Wondering if being aware of the context is responsibility of the masking component or instead the target component should be responsible to register the secrets?

Basically, I would like to reduce the coupling of this masking component; ideally, it should only depend on logging.